bson/bsoncodec/string_codec.go

```diff
@@ -72,6 +72,12 @@ func (sc *StringCodec) DecodeValue(dctx DecodeContext, vr bsonrw.ValueReader, va
 		if err != nil {
 			return err
 		}
+	case bsontype.Binary:
```

Comment:
Is this something we missed in the string codec ticket?

Reply:
It turns out mgo doesn't do this and I left it out for a reason